src/coreclr/vm/frozenobjectheap.cpp

```diff
@@ -137,15 +137,15 @@ Object* FrozenObjectSegment::TryAllocateObject(PTR_MethodTable type, size_t obje
     _ASSERT(m_pStart != nullptr && FOH_SEGMENT_SIZE > 0 && m_SegmentHandle != nullptr); // Expected to be inited
     _ASSERT(IS_ALIGNED(m_pCurrent, DATA_ALIGNMENT));
 
-    uint8_t* obj = m_pCurrent;
-    if (reinterpret_cast<size_t>(m_pStart + FOH_SEGMENT_SIZE) < reinterpret_cast<size_t>(obj + objectSize))
+    const size_t nextObject = reinterpret_cast<size_t>(m_pCurrent + objectSize);
+    if (reinterpret_cast<size_t>(m_pStart + FOH_SEGMENT_SIZE) < nextObject)
```

Comment:
Nit: These casts to `size_t` do not look necessary. Comparing the `uint8_t*` pointers should work just fine.